programs/protocol_event/src/state/event.rs

```diff
@@ -26,45 +28,12 @@ impl Event {
     const MAX_EVENT_NAME_LENGTH: usize = 50;
     const MAX_PARTICIPANTS: usize = 300;
 
-    pub const SIZE: usize =
-        DISCRIMINATOR_SIZE
-            + (PUB_KEY_SIZE * 2)
-            + Category::SIZE
-            + EventGroup::SIZE
-            + BOOL_SIZE
-            + vec_size(CHAR_SIZE, Event::MAX_EVENT_SLUG_LENGTH)
-            + vec_size(CHAR_SIZE, Event::MAX_EVENT_NAME_LENGTH)
-            + U64_SIZE
-            + vec_size(U64_SIZE, Event::MAX_PARTICIPANTS)
-            + option_size(I64_SIZE) * 2;
-}
-
-#[derive(AnchorSerialize, AnchorDeserialize, Debug, Clone, PartialEq)]
-pub struct Category {
-    pub id: String, // category identifier e.g. FOOTBALL
-    pub name: String, // for display purposes e.g Football
-}
-
-impl Category {
-    const MAX_ID_LENGTH: usize = 10;
-    const MAX_CATEGORY_NAME_LENGTH: usize = 25;
-
-    const SIZE: usize =
-        vec_size(CHAR_SIZE, Category::MAX_ID_LENGTH)
-        + vec_size(CHAR_SIZE, Category::MAX_CATEGORY_NAME_LENGTH);
-}
-
-#[derive(AnchorSerialize, AnchorDeserialize, Debug, Clone, PartialEq)]
-pub struct EventGroup {
-    pub id: String, // event group identifier e.g. MLS
-    pub name: String, // for display purposes e.g. Major League Soccer
-}
-
-impl EventGroup {
-    const MAX_ID_LENGTH: usize = 10;
-    const MAX_EVENT_GROUP_NAME_LENGTH: usize = 25;
-
-    const SIZE: usize =
-        vec_size(CHAR_SIZE, EventGroup::MAX_ID_LENGTH)
-            + vec_size(CHAR_SIZE, EventGroup::MAX_EVENT_GROUP_NAME_LENGTH);
+    pub const SIZE: usize = DISCRIMINATOR_SIZE
+        + (PUB_KEY_SIZE * 4)
+        + BOOL_SIZE
+        + vec_size(CHAR_SIZE, Event::MAX_EVENT_SLUG_LENGTH)
+        + vec_size(CHAR_SIZE, Event::MAX_EVENT_NAME_LENGTH)
+        + U64_SIZE
+        + vec_size(U64_SIZE, Event::MAX_PARTICIPANTS)
```

Comment:
Might as well use the correct type size for these,`I64_SIZE` even if they are equivalent.

Reply:
The participants are a vec of u16 so I'm not sure where I got u64 from